examples/extension-demo/src/ExamplePanel.tsx

```diff
@@ -0,0 +1,122 @@
+import { CompressedImage } from "@foxglove/schemas/schemas/typescript";
+import { PanelExtensionContext, RenderState, Topic, MessageEvent } from "@foxglove/studio";
+import { useLayoutEffect, useEffect, useState, useRef, useMemo } from "react";
+import ReactDOM from "react-dom";
+
+type ImageMessage = MessageEvent<CompressedImage>;
+
+type PanelState = {
+  topic?: string;
+};
+
+// Draws the compressed image data into our canvas.
+async function drawImageOnCanvas(imgData: Uint8Array, canvas: HTMLCanvasElement, format: string) {
+  const ctx = canvas.getContext("2d");
+  if (ctx == undefined) {
+    return;
+  }
+
+  // Create a bitmap from our raw compressed image data.
+  const blob = new Blob([imgData], { type: `image/${format}` });
+  const bitmap = await createImageBitmap(blob);
+
+  // Adjust for aspect ratio.
+  canvas.width = Math.round((canvas.height * bitmap.width) / bitmap.height);
+
+  // Draw the image.
+  ctx.drawImage(bitmap, 0, 0, canvas.width, canvas.height);
+  ctx.resetTransform();
+}
+
+function ExamplePanel({ context }: { context: PanelExtensionContext }): JSX.Element {
+  const [topics, setTopics] = useState<readonly Topic[] | undefined>();
+  const [message, setMessage] = useState<ImageMessage>();
+
+  const [renderDone, setRenderDone] = useState<(() => void) | undefined>();
+
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+
+  // Restore our state from the layout via the context.initialState property.
+  const [state, setState] = useState<PanelState>(() => {
+    return context.initialState as PanelState;
+  });
+
+  // Filter all of our topics to find the ones with a CompresssedImage message.
+  const imageTopics = useMemo(
+    () => (topics ?? []).filter((topic) => topic.datatype === "sensor_msgs/CompressedImage"),
+    [topics],
+  );
+
+  // Subscribe to the new image topic when a new topic is chosen.
+  useEffect(() => {
+    if (state.topic) {
+      context.subscribe([state.topic]);
+    }
+  }, [context, state.topic]);
+
+  // Choose our first available image topic as a default once we have a list of topics available.
+  useEffect(() => {
+    if (state.topic == undefined) {
+      setState({ topic: imageTopics[0]?.name });
+    }
+  }, [state.topic, imageTopics]);
+
+  // Save our state to the layout when the topic changes.
+  useEffect(() => {
+    context.saveState({ topic: state.topic });
+  }, [state.topic]);
```

Comment:
@foxymiles - confirming that we need `context` as a dependency here / this `useEffect` can be combined with the one on line 51?

Reply:
This would apply to the existing image extension panel example as well.

Reply:
Yes you're right. I think we still need to set up the eslint rules for hooks for this repo. In practice `context` won't change but for correctness we should include it in the dependencies list.

Reply:
Pushed follow-up commit; will open separate PR for the existing image extension panel example.